### ISSUE TITLE
hyperparameters setting problem

### ISSUE THREAD

**Issue by shuaiNJU** [amazon-research/long-tailed-ood-detection#5]
Hi,
I have a confusion about the hyperparameters setting problem. In your original paper, the hyperparameter setting is  “On all datasets, we set τ = 0.1 following (Khosla et al., 2020), λ1 = 0.5 following (Hendrycks et al.,2019), and empirically set λ2 = 0.1 for PASCL”, and for auxiliary  branch finetuning, the eopch is 3. However ,on the code, for CIFAR10-LT，“--Lambda2 0.1 --T 0.07”， for CIFAR100-LT，“--Lambda2 0.02 --T 0.2”, and for auxiliary  branch finetuning, the eopch is 10.  So I want to know which hyperparameter values can be used to reproduce the right results?
Thanks a lot!

**Comment by htwang14** [comment]
The paper documented the original hyper-parameters I used to get the reported results. The settings suggested in the repo may lead to slightly better results. But most likely the difference would be insignificant.

**Comment by anohdnu** [comment]
Hi,
How did you set other hyperparameters like no of training epochs, learning rate, and eventually choose the best model. Since stage 1 training script there is no validation set used and evaluation is observed for test set itself. Were you used an additional validation set?

**Comment by htwang14** [comment]
> Hi, How did you set other hyperparameters like no of training epochs, learning rate, and eventually choose the best model. Since stage 1 training script there is no validation set used and evaluation is observed for test set itself. Were you used an additional validation set?

Those hyperparameters (epochs, lr, batch size, optimizer choices, etc) are all documented in Appendix A of the [paper](https://proceedings.mlr.press/v162/wang22aq/wang22aq.pdf).

To choose the model in both stage 1 and 2, I used the last epoch model, which is saved as `latest.pth` [here](https://github.com/amazon-science/long-tailed-ood-detection/blob/e578f44f743a958187ac30180bc168a0d99f207a/stage1.py#L382) for stage 1 and [here](https://github.com/amazon-science/long-tailed-ood-detection/blob/e578f44f743a958187ac30180bc168a0d99f207a/stage2.py#L353) for stage 2.

More specifically, after stage 1, you get a model named `latest.pth` from [this line of code](https://github.com/amazon-science/long-tailed-ood-detection/blob/e578f44f743a958187ac30180bc168a0d99f207a/stage1.py#L382).

Then the above model is used as the initialization for stage 2, which is loaded using [this line of code](https://github.com/amazon-science/long-tailed-ood-detection/blob/e578f44f743a958187ac30180bc168a0d99f207a/stage2.py#L203).

Then after stage 2 training is done, you get another model also named `latest.pth` (but saved in a subfolder) from [this line of code](https://github.com/amazon-science/long-tailed-ood-detection/blob/e578f44f743a958187ac30180bc168a0d99f207a/stage2.py#L353), which is used for the final testing.

**Comment by anohdnu** [comment]
Thank you. Yes I saw the values for epochs, lr, batch size, etc you have selected in Appendix A. But how did you selected these values. e.g. 200 epochs are good for CIFAR , based on test performance or was there a seperate validation set you used.
